### ISSUE TITLE
issues with Denizen plugin

### ISSUE THREAD

**Issue by Jabelpeeps** [PaperMC/Paper#274]
Hi, 

I'm not sure if you would call this a crash or a bug, so I've answered the questions from both templates in the hope that it will be helpful to you.
# Incompatibility (Bug)
### What behaviour is expected:

Using the Denizen scripting plugin (part of the citizens2 collection), it is possible to add command scripts to the game.  Running on upstream spigot these work as designed, I was expecting Paper to behave the same.
### What behaviour is observed:

The commands do not work, and an error is produced on the console (copied in below)
### Steps/models to reproduce:

In the attached stack trace, the command I was attempting to run is from the script on this page:- http://mcmonkey.org/denizen/repo/entry/69
### Link/paste of stack trace

```
[20:37:02 WARN]: Unexpected exception while parsing console command "holo"
java.lang.NullPointerException
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:152) ~[paper.jar:git-Paper-726]
        at org.bukkit.craftbukkit.v1_9_R1.CraftServer.dispatchCommand(CraftServer.java:664) ~[paper.jar:git-Paper-726]
        at org.bukkit.craftbukkit.v1_9_R1.CraftServer.dispatchServerCommand(CraftServer.java:627) [paper.jar:git-Paper-726]
        at net.minecraft.server.v1_9_R1.DedicatedServer.aL(DedicatedServer.java:442) [paper.jar:git-Paper-726]
        at net.minecraft.server.v1_9_R1.DedicatedServer.D(DedicatedServer.java:405) [paper.jar:git-Paper-726]
        at net.minecraft.server.v1_9_R1.MinecraftServer.C(MinecraftServer.java:723) [paper.jar:git-Paper-726]
        at net.minecraft.server.v1_9_R1.MinecraftServer.run(MinecraftServer.java:622) [paper.jar:git-Paper-726]
        at java.lang.Thread.run(Thread.java:745) [?:1.8.0_91]
```
### Plugin list:

AdminAnything_Free, PlaceholderAPI, LagCheck, AutoShutdown, PorteCoulissante, ProtocolLib, FastAsyncWorldEdit, ExecuteEverywhere, Nicky, VoxelSniper, BungeePermsBukkit, BlocksHub, RetractableBridge, Vault, BungeeTabListPlus, Porty-Bukkit, WorldEdit, VentureChat, GiantTrees, LogBlock, DuckLog, WorldGuard, Citizens, BlocksHub-WorldGuard, BiomeGenerator, BlocksHub-LogBlock, AncientGates, Denizen, Depenizen
### Paper build number:

`This server is running Paper version git-Paper-726 (MC: 1.9.2) (Implementing API version 1.9.2-R0.1-SNAPSHOT)`

**Comment by zachbr** [comment]
This should already be caught and handled in this version so I'm immediately hesitant about whether you are truly on 726 or not.

Regardless, your reproduction instructions are not sufficient.

**Comment by Jabelpeeps** [comment]
As far as I can see, the paper build number is at the end of every line in the stack trace, I'm not sure how you can doubt the version of it.

Putting that aside, what information are you looking for me to add in the steps to reproduce?

Basically, I booted the server, with the plugins in the list above, and tried to run the command '/holo'.

Are you looking for me to post the contents of any of the config files?  I'm happy to do so, just let me know which files you want to see.

If you are not familiar with the Denizen plugin, I can post a link to it (the source code is here on github.)

**Comment by zachbr** [comment]
@Jabelpeeps we have had a string of people lying about their version info in order to get support or otherwise be lazy. Apologies, little over-stressed at the moment.

As far as the reproduction instructions go, yes I need you to be stupid specific on how denizen is setup. ELI5, or some random noobie on minecraft forums. Specifically configuration. Many of our contributors have not ever touched Denizen and being as specific as possible allows as many people as possible to get involved.

To get back to the issue at hand, I have not had any free time to even begin looking into this issue, nor do I expect to get any for at least a few days.

**Comment by Jabelpeeps** [comment]
@Zbob750 I forgive you for your doubt over my version, I know well how frustrating it can be when you think you've fixed something and it comes back in again.  I hope you get your stress lessened...

As for reproducing the bug..., I did another test, with only Denizen installed, to exclude the complications of the other plugins

I made a fresh folder, and copied 1.9.2 versions of spigot and paper to the folder (both newly downloaded/compiled on the same day).  I booted the spigot .jar first, accepted the eula, but otherwise left the config unchanged.  Once the worlds had generated, I shut it down, and booted the paper .jar (copied from the cache folder of a paperclip install.)  So far no problems, and I made no changes to paper.yml.

I then copied the denizen.jar into the plugins folder;  the version of Denizen that I used is the latest one available at the time of writing, here is a link to the ci server page for the plugin:-  http://ci.citizensnpcs.co/job/Denizen/1615/  
I booted the server another time with each of spigot and paper, with no errors being apparent - but no extra functions from denizen, as no scripts were installed; again I left the default config unchanged.

I install the same script that I linked to in the OP, copied into the scripts folder, inside the plugins/Denizen folder.

Here is the console log when I booted with spigot, and tried the command 'holo' from the console:- http://pastebin.com/BEvbwsvk
the command does not work, but the error message is the one specified in the script for when the commands are attempts from the console, therefore it appears to parsing the command and responding as designed.

I then booted the server with paper, having changed nothing in the configuration, and tried the same command 'holo';  here is a link to the console log:-  http://pastebin.com/hHAkG14D

For good measure, I replaced the denizen script with another one from the offical script repo, and again spigot behaved as expected, but paper gave a very similar looking NPE, when I tried to use one of the commands specified in the script.

I hope this is fixable, I'm pretty happy with the enhancements of paper, but denizen is essential to a project we are building, so I've had to put that server back onto spigot for now.

Let me know if you need any further details.

many thanks.

**Comment by zachbr** [comment]
Thanks for providing more detail, we'll look at this as soon as we can.

**Comment by Jabelpeeps** [comment]
Do I loose my 'Incomplete' label now? =)

**Comment by zachbr** [comment]
Woops

**Comment by zachbr** [comment]
Fixed in 1.9.4. Which of course Denizens doesn't as of yet have a build for, but I assume will shortly being that it's such a minor update.

I'd say the actual build numbers it's fixed in but uhh, CI is down because the host is doing network maintenance for the next 20 minutes. Safe to say it'll be in whatever build you can get once its back up.

Edit: Builds 739+ will have this included
